src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -75,6 +77,23 @@ public HttpRequestInterceptionBuilder()
     /// </remarks>
     internal string? Key => _revision == _matchKeyRevision ? _matchKey : null;
 
+    /// <summary>
+    /// Configures the builder to match a request that meets one of several criteria.
+    /// </summary>
+    /// <param name="predicate">
+    /// A delegate to an asynchronous method which returns <see langword="true"/> if
+    /// the request is considered a match; otherwise <see langword="false"/>.
+    /// </param>
+    /// <returns>
+    /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
+    /// </returns>
+    public HttpRequestInterceptionBuilder AndFor(Func<bool>? predicate)
```

Comment:
One possibility would be to take a `params Func<HttpRequestMessage, Task<bool>>[]` and then the user could specify multiple delegates and then they're combined into one here before being assigned to `_requestMatcher`.

Something like the below, after having validated there's at least two delegates in the array.

There might be a smarter way to do it with combining the delegates, rather than having to capture them into a Lambda and iterate them explicitly.

```csharp
_requestMatcher = async (p) =>
{
    foreach (predicate in predicates)
    {
        if (!await predicate(p))
        {
            return false;
        }
    }

    return true;
};
```